Modules/IO/NIFTI/test/itkNiftiImageIOTest14.cxx

```diff
@@ -90,7 +90,7 @@ itkNiftiImageIOTest14(int argc, char * argv[])
     }
   }
 
-  const char * output_test_fn = "xyzt_units_output_check.nii";
+  const std::string output_test_fn = std::string(test_dir) + "/xyzt_units_output_check.nii";
```

Comment:
Probably cleaner to use: `std::filesystem::path::preferred_separator` or a `std::filesystem::path` object to concatenate paths and then get the `std::string` from it. I think both require C++17, which we do.

Reply:
`std::filesystem` isn't really used in ITK yet is it? Seems too close to the 5.4 release to start using it. Yes, C++17 is already required, but using `std::filesystem` also makes requirements of the system' STL library...

Reply:
>  but using std::filesystem also makes requirements of the system' STL library

Not sure if that would pose a problem. But fair enough, the issue may be considered after 5.4, as PR #4562 makes progress.

Feel free to resolve the conversation and merge the PR then.

Reply:
Thanks for these comments, good to know the future plan but will leave as-is for now then

Reply:
>Not sure if that would pose a problem. But fair enough, the issue may be considered after 5.4, as PR https://github.com/InsightSoftwareConsortium/ITK/pull/4562 makes progress.

In fact, I looked into it, and I think it would force dropping support for macOS <= 10.14, which would be a shame:

https://stackoverflow.com/questions/58667853/does-use-of-c17-stdfilesystem-require-macos-10-15-xcode-11-1

But indeed, that's for https://github.com/InsightSoftwareConsortium/ITK/pull/4562 now...

Reply:
> Probably cleaner to use: `std::filesystem::path::preferred_separator` or a `std::filesystem::path` object to concatenate paths and then get the `std::string` from it. I think both require C++17, which we do.

itksys::SystemTools::JoinPath is the current way.

Reply:
Thanks @blowekamp I will update

Reply:
Thanks for looking into this issue.